### PR TITLE
OCM-4442: Moved kmsARN regexp validator to common lib

### PR DESCRIPTION
**Pull request by den-rgb**
moved kms Arn regexp validator to common lib
Related to : https://github.com/openshift-online/ocm-common/pull/13
JIRA: https://issues.redhat.com/browse/OCM-3155, https://issues.redhat.com/browse/OCM-4442

### CODE REVIEW

File: internal/ocm/resource/cluster.go [hunk 1]
```diff
@@ -159,6 +155,15 @@ func (c *Cluster) CreateAWSBuilder(awsTags map[string]string, ec2MetadataHttpTok
 	return nil
 }
 
+func (c *Cluster) ProcessKMSKeyARN(kmsKeyARN *string, awsBuilder *cmv1.AWSBuilder) error {
+	err := kmsArnRegexpValidator.ValidateKMSKeyARN(kmsKeyARN)
```
[the remark this answers is not in the document]
Comment (den-rgb) [reply internal/ocm/resource/cluster.go:159]: do you mean separating the `aws.builder` and the validator and having the validator up near the beginning of `CreateAWSBuilder` or moving the validator before this function call: https://github.com/terraform-redhat/terraform-provider-rhcs/blob/main/provider/clusterrosaclassic/cluster_rosa_classic_resource.go#L631 ?

**Comment by sagidayan** [comment]
@den-rgb please rebase your branch to include latest CI changes due to tf1.6.0 breaking changes.

Also before pushing, run `make tests` and ensure they all pass

**Comment by den-rgb** [comment]
> make tests

I need to wait for ocm-common changes to be merged first before I can continue with this

**Comment by nirarg** [comment]
@den-rgb thank you for this contribution
I'm putting this PR on hold because next version v1.4.0 is in code freeze
I will release the hold once the main branch will open to v1.5.0
/hold

**Comment by nirarg** [comment]
/hold cancel
The main branch is alredy opened for v1.5.0
@den-rgb Do you think it ready for merge?

**Comment by den-rgb** [comment]
> /hold cancel The main branch is alredy opened for v1.5.0 @den-rgb Do you think it ready for merge?

Yes It should be ready to merge, I'll just rebase it just in case and if everything is alright then it should be good to go

**Comment by den-rgb** [comment]
Could someone point me to how to resolve these errors as they don't seem to be related to my changes, Thanks

**Comment by bardielle** [comment]
can you rebace on the main branch

**Comment by sagidayan** [comment]
/approve

Thanks @den-rgb 👏🏽

**Pull request merged**